examples/antarctic_slope_current_model.jl

```diff
@@ -0,0 +1,154 @@
+
+# Linear near-surface generator
+ζ₀(k) = 1 + (h(k) - 1) / refinement
+
+# Bottom-intensified stretching function
+Σ(k) = (1 - exp(-stretching * h(k))) / (1 - exp(-stretching))
+
+# Generating function
+z_faces(k) = Lz * (ζ₀(k) * Σ(k) - 1)
+
+underlying_grid = RectilinearGrid(arch,
+                                  size = (Nx, Ny, Nz),
+                                  topology = (Periodic, Bounded, Bounded),
+                                  x = (-Lx/2, Lx/2),
+                                  y = (0, Ly),
+                                  z = z_faces)
+
+println(underlying_grid)
+
+# We want the underwater slope to provide a depth of 500 m at y = 0 and the full 4 km by y =200. It follows
+# a hyperbolic tangent curve with its center at y ~= 150 at a depth of ~ -4000 + (4000 - 500)/2 = -2250 m
+underwater_slope(x, y) = -1750tanh(0.00004*(y - 150000)) - 2250
+# TODO: add 50km troughs to the slope, dependent on x and y. Use appendix C to add detail here
+
+grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(underwater_slope))
+
+println(grid)
+
+# TODO: add underwater slope at y = 0 with troughs
+
+#
+# Creating the Hydrostatic model:
+#
+
+# Forcings:
+u_forcing(x, y, z, t) = exp(z) * cos(x) * sin(t) # Not actual forcing, just example
+
+# TODO: need to use forcings to enact the sponge layers. Their support is within 20000 m
+# of the N/S boundaries, they impose a cross-slope buoyancy gradient, and the relaxation
+# tiem scales decrease linearly with distance from the interior termination of the sponge layers
+
+# We'll use Relaxation() to impose a sponge layer forcing on velocity, temperature, and salinity
+damping_rate       = 1 / 43200 # Relaxation time scale in seconds, need to make this decrease linearly toward outermost boundary
+south_mask         = GaussianMask{:y}(center=0, width=sponge_width)
+north_mask         = GaussianMask{:y}(center=Ly, width=sponge_width)
+south_sponge_layer = Relaxation(; rate=damping_rate, mask=south_mask)
+north_sponge_layer = Relaxation(; rate=damping_rate, mask=north_mask)
+sponge_layers      = (south_sponge_layer, north_sponge_layer)
+# TODO: compose north_mask and south_mask together into one sponge layer, OR compose north/south sponge layers
+
+
+# Boundary Conditions:
+no_slip_bc   = ValueBoundaryCondition(0.0)
+free_slip_bc = FluxBoundaryCondition(nothing)
+
+free_slip_surface_bcs = FieldBoundaryConditions(no_slip_bc, top=FluxBoundaryCondition(nothing))
+no_slip_field_bcs     = FieldBoundaryConditions(no_slip_bc)
+
+# Buoyancy Equations of State - we want high order polynomials, so we'll use TEOS-10
+eos = TEOS10EquationOfState()
+
+# Coriolis Effect, using basic f-plane with precribed reference Coriolis parameter
+coriolis = FPlane(f=-1.3e14)
+
+# Diffusivities as part of closure
+# TODO: make sure this works for biharmonic diffusivities as the horizontal, 
+horizontal_closure = HorizontalScalarDiffusivity(ν=0.1, κ=0.1)
+vertical_closure   = VerticalScalarDiffusivity(ν=3e-4, κ=1e-5)
```

Comment:
We don't need biharmonic or a background diffusivity if we use WENO + CATKE, so we can eliminate these